tests/test_job_router.py

```
@@ -124,9 +117,49 @@ def test_successful_route_returns_worker_id():
+    print("PASSED")
+
+
+def test_route_raises_clear_error_on_malformed_ring_metadata():
+    print("\n--- TEST 6: malformed ring metadata raises RuntimeError, not KeyError ---")
+    ring = ConsistentHashRing()
+    ring.add_worker("worker-1", {"address": "worker-1"})  # missing "port"
+    router = JobRouter(ring)
+
+    try:
+        router.route("job-1", "echo hi")
+        assert False, "expected RuntimeError to be raised"
```

Ruff (0.16.1)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)